packages/core/src/html/BootstrapVueProcessor.js

```diff
@@ -0,0 +1,167 @@
+  }
+
+  /**
+   * Takes an element, looks for direct elements with slots and transforms to avoid Vue parsing.
+   * This is so that we can use bootstrap-vue popovers, tooltips, and modals.
+   * @param node Element to transform
+   */
+  static transformSlottedComponents(node) {
+    node.children.forEach((child) => {
+    // Turns <template #content>... into <span data-mb-slot-name=content>...
+      const vslotShorthandName = getVslotShorthandName(child);
+      if (vslotShorthandName) {
+        child.attribs['data-mb-slot-name'] = vslotShorthandName;
+        delete child.attribs[`#${vslotShorthandName}`];
+        // similarly, need to transform templates to avoid Vue parsing
+        child.name = 'span';
+      }
+    });
+  }
+
+  /*
+   * Popovers, tooltips, modals, triggers
+   *
+   * We use bootstrap-vue's popovers, tooltips and modals, but perform various transformations
+   * to conform with our syntax instead, and to support triggers.
+   *
+   * For tooltips and popovers,
+   * The content / title is put into hidden [data-mb-slot-name] slots.
+   * Then, we call the relevant content getters inside core-web/index.js at runtime to get this content.
+   *
+   * For modals,
+   * only syntactic transformations are performed.
+   *
+   * For triggers,
+   * When building the site, we can't immediately tell whether a trigger references
+   * a modal, popover, or tooltip, as the element may not have been processed yet.
+   *
+   * So, we make every trigger try all 3 at runtime in the browser. (refer to Trigger.vue)
+   * It will attempt to open a modal, then a tooltip or popover.
+   * For modals, we simply attempt to show the modal via bootstrap-vue's programmatic api.
+   * The content of tooltips and popovers is retrieved from the [data-mb-slot-name] slots,
+   * then the <b-popover/tooltip> component is dynamically created appropriately.
+   */
+
+  static addTriggerClass(node, trigger) {
+    const triggerClass = trigger === 'click' ? 'trigger-click' : 'trigger';
+    node.attribs.class = node.attribs.class ? `${node.attribs.class} ${triggerClass}` : triggerClass;
+  }
+
+  static _renameSlot(node, originalName, newName) {
+    if (node.children) {
+      node.children.forEach((child) => {
+        const vslotShorthandName = getVslotShorthandName(child);
+        if (vslotShorthandName && vslotShorthandName === originalName) {
+          const newVslot = `#${newName}`;
+          child.attribs[newVslot] = '';
+          delete child.attribs[`#${vslotShorthandName}`];
+        }
+      });
+    }
+  }
+
+  static _renameAttribute(node, originalAttribute, newAttribute) {
+    if (_.has(node.attribs, originalAttribute)) {
+      node.attribs[newAttribute] = node.attribs[originalAttribute];
+      delete node.attribs[originalAttribute];
+    }
+  }
+
+  processPopover(node) {
+    warnDeprecatedAttributes(node, { title: 'header' });
+
+    this.markdownProcessor.processAttributeWithoutOverride(node, 'content', true);
```

Comment:
could move these to `ComponentProcessor` so this file actually only handles the bootstrap vue transformations; similarly for the warnings

Reply:
Okay, I have moved the warnings to `warning.js` and the rendering of markdown-in-attributes to `ComponentProcessor`